docs/src/main/asciidoc/adapters/gcp-intro.adoc

```diff
@@ -66,64 +111,166 @@ curl http://localhost:8080/ -d "hello"
+
+===== Getting Started
+
+Let’s start with a simple Spring Cloud Function which will run as a GCF background function:
+
+[source, java]
+----
+@SpringBootApplication
+public class BackgroundFunctionMain {
+
+	public static void main(String[] args) {
+		SpringApplication.run(BackgroundFunctionMain.class, args);
+	}
+
+	@Bean
+	public Consumer<PubSubMessage> pubSubFunction() {
+		return message -> System.out.println("The Pub/Sub message data: " + message.getData());
+	}
+}
+----
+
+In addition, create `PubSubMessage` class in the project with the below definition.
+This class represents the https://cloud.google.com/functions/docs/calling/pubsub#event_structure[Pub/Sub event structure] which gets passed to your function on a Pub/Sub topic event.
+
+[source, java]
+----
+public class PubSubMessage {
+
+	private String data;
+
+	private Map<String, String> attributes;
+
+	private String messageId;
+
+	private String publishTime;
+
+	public String getData() {
+		return data;
+	}
+
+	public void setData(String data) {
+		this.data = data;
+	}
+
+	public Map<String, String> getAttributes() {
+		return attributes;
+	}
+
+	public void setAttributes(Map<String, String> attributes) {
+		this.attributes = attributes;
+	}
+
+	public String getMessageId() {
+		return messageId;
+	}
+
+	public void setMessageId(String messageId) {
+		this.messageId = messageId;
+	}
+
+	public String getPublishTime() {
+		return publishTime;
+	}
+
+	public void setPublishTime(String publishTime) {
+		this.publishTime = publishTime;
+	}
+
+}
+----
+
+Specify your configuration main class in `resources/META-INF/MANIFEST.MF`.
+
+[source]
+----
+Main-Class: com.example.BackgroundFunctionMain
+----
+
+Then run the function locally.
+This is provided by the Google Cloud Functions `function-maven-plugin` described in the project dependencies section.
+
+----
+mvn function:run
+----
+
+Invoke the HTTP function:
+
+----
+curl localhost:8080 -H "Content-Type: application/json" -d '{"data":"hello"}'
+----
+
+Verify that the function was invoked by viewing the logs.
+
+===== Deploy to GCP
+
+In order to deploy your background function to GCP, first package your application.
 
 ----
 mvn package
 ----
 
-You should see the resulting JAR in `target/deploy` directory.
+If you added the custom `spring-boot-maven-plugin` plugin defined above, you should see the resulting JAR in `target/deploy` directory.
 This JAR is correctly formatted for deployment to Google Cloud Functions.
 
-Make sure that you have the https://cloud.google.com/sdk/install[Cloud SDK CLI] installed.
+Next, make sure that you have the https://cloud.google.com/sdk/install[Cloud SDK CLI] installed.
 
 From the project base directory run the following command to deploy.
 
 ----
-gcloud alpha functions deploy function-sample-gcp \
+gcloud alpha functions deploy function-sample-gcp-background \
```

Comment:
I am wondering if `--allow-unauthenticated` should be added

Reply:
Good question - I am thinking of leaving out because the user will get an explicit [yes/no] prompt for whether they want unauthenticated invocations if they run this command without the extra argument. Maybe this will be more clear to avoid accidents with security